ADAL/src/ADAggregatedDispatcher.m

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "ADTelemetry.h"
+#import "ADEventInterface.h"
+#import "ADAggregatedDispatcher.h"
+
+@implementation ADAggregatedDispatcher
+
+- (id)init
+{
+    //Ensure that the appropriate init function is called. This will cause the runtime to throw.
+    [super doesNotRecognizeSelector:_cmd];
+    return nil;
+}
+
+- (id)initWithDispatcher:(id<ADDispatcher>)dispatcher
+{
+    self = [super initWithDispatcher:dispatcher];
+    return self;
+}
+
+- (void)flush
+{
+    // Integrate events of a particular request id into one single event
+    for (NSString* requestId in _objectsToBeDispatched)
```

Comment:
_objectsToBeDispatched should be protected by a lock of some kind to make sure no one changes it while you're using it.

For this sort of "flush" operation the most performant thing you can do is store the old value in a local variable and replace it with a new array so that you minimize the amount of time you're holding the lock. e.g.:

```
[lock grab];
NSMutableArray* localObjects = _objectsToBeDispatched;
_objectsToBeDispatched = [NSMutableArray new];
[lock release];

for (NSString* requestId in localObjects) { // do stuff }
SAFE_ARC_RELEASE(localObjects);